nc/nc.go

```diff
@@ -97,13 +98,24 @@ type AcknowledgementPayload struct {
 }
 
 type NotificationClient interface {
-	Start(endpoint string, clientID string, username string, password string) error
+	Start() error
 	Stop() error
 }
 
+type ClientBuilder interface {
+	Build(opts *mqtt.ClientOptions) mqtt.Client
+}
+
+type MqttClientBuilder struct{}
+
+func (MqttClientBuilder) Build(opts *mqtt.ClientOptions) mqtt.Client {
+	return mqtt.NewClient(opts)
+}
+
 // Client is a client for Notification center
 type Client struct {
-	client mqtt.Client // nil if not started or stopped, always check for nil before using
+	clientBuilder ClientBuilder
```

Comment:
Why is this added? Any reasons creating it with `mqtt.NewClient(opts)` when needed is not enough?

Reply:
We have to reinitialize the client when reconnecting https://github.com/NordSecurity/nordvpn-linux/pull/213/files#diff-3ef8346631bf7e381d05cd8c7af5211833424bae0c8a08f8b79831acf2ff0755R217. This function is unit tested, so we have to use a mock so that we do not build a real mqtt client.